plugin/xcodebuild.vim

```diff
@@ -26,7 +26,7 @@ function! s:test()
 endfunction
 
 function! s:run_command(cmd)
-  execute '!' . a:cmd
+  execute s:runner() . ' ' . a:cmd
```

Comment:
Big :+1: on adding support for configurable and async runners. This form will certainly work for the listed runners (shell, tmux-runner, and dispatch) as they all provide commands, but in some cases a user might want to run via a Vim function as the plugin they use only provides functions. Likewise, although you don't need it now, a template string approach can grow to support more options later if needed.
 
You can see the relevant examples in [vim-rspec](https://github.com/thoughtbot/vim-rspec/blob/2cf8c5002cb89eb8e9aabe59d25b89dc5afec88a/plugin/rspec.vim#L43-L47) and a more complex version in [vim-spec-runner](https://github.com/gabebw/vim-spec-runner/blob/2550392842e16a512fc42ef7ee761ad7180cf20e/plugin/spec-runner.vim#L165-L178).

This would likely mean having a command template string as the configuration point, rather than `g:xcodebuild_runner`. Something like:

``` vim
let  s:default_run_command = '!'

function! s:run_command(cmd)
  let run_cmd = substitute(s:runner_template(), "{cmd}", a:cmd, "g")
  execute run_cmd
endfunction

function! s:runner_template()
  if exists('g:xcodebuild_run_command')
    return g:xcodebuild_run_command
  else
    return s:default_run_command
  endif
endfunction
```